### PR TITLE
Convert memory_vm_chart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

**Comment by polina-c** [comment]
blocked by memory_analyzer migration

File: packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart [hunk 1]
```diff
@@ -127,6 +127,9 @@ class MemoryVMChartState extends State<MemoryVMChart> with AutoDisposeMixin {
   void didChangeDependencies() {
     super.didChangeDependencies();
 
+    if (_initialized) return;
+    _initialized = true;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart:131]: We don't want to return early if we are initialized - we just want to return early if the new controller from provider is the same we already have. Follow the pattern here: https://github.com/flutter/devtools/blob/85a45e5fd7cdf5946da19611dd18513bc28b22bb/packages/devtools_app/lib/src/screens/profiler/profiler_screen.dart#L97-L100

Also, make sure we are calling any code not dependent upon the memory controller before we return early if memoryController has not changed (in this case, set colorScheme before returning early).

File: packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart [hunk 2]
```diff
@@ -23,17 +21,17 @@ class VMChartController extends ChartController {
           name: 'VM Memory',
         );
 
-  final MemoryController _memoryController;
+  late final MemoryController _memoryController;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart:24]: done

File: packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart [hunk 3]
```diff
@@ -23,17 +21,17 @@ class VMChartController extends ChartController {
...
 
   // TODO(terry): Only load max visible data collected, when pruning of data
   //              charted is added.
   /// Preload any existing data collected but not in the chart.
   @override
   void setupData() {
     final chartDataLength = timestampsLength;
-    final dataLength = _memoryController.memoryTimeline.data.length;
+    final dataLength = _memoryController.memoryTimeline!.data.length;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart:32]: done

File: packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart [hunk 4]
```diff
@@ -106,15 +104,17 @@ enum TraceName {
...
 
   /// Controller for managing memory collection.
-  MemoryController _memoryController;
+  late MemoryController _memoryController;
 
-  MemoryTimeline get _memoryTimeline => _memoryController.memoryTimeline;
+  MemoryTimeline? get _memoryTimeline => _memoryController.memoryTimeline;
 
-  ColorScheme colorScheme;
+  ColorScheme? colorScheme;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart:117]: removed as it is not used

File: packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart:131]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**